### ISSUE TITLE
Ruby 1.9.2 key issue

### ISSUE THREAD

**Issue by turlockmike** [Soleone/sc2ranks#1]
The character class is attempting to access each member of the hash by symbol, however, this causes problems in ruby 1.9.2. The way i solved was by changing line 17 of the sc2ranks.rb file to 

```
self[member] = hash[member.to_s]
```

**Comment by Soleone** [comment]
Thanks for the report, I will make sure to change it. You're the first official user I think :D

Does the library appear useful at this point yet? Anything you would like to see implemented first?

**Comment by turlockmike** [comment]
Yeah it works fine. I made some changes in order to make it work for my application specifically. I added some code in order to get information about teams.

```
def find_teams(name, code = nil, region = REGIONS.first)
  if !code
    characters = search(name, region)
    code = characters.first.bnet_id
  end
  advanced_character(name, code, region)
end

def advanced_character(name, code, region = REGIONS.first)
  result = get("/base/teams/#{region}/#{encoded_name(name, code)}")
  Character.new(result.parsed_response)
end
```

which also required me to change the character class to get the information i want.

```
class Character < Struct.new(:name, :bnet_id, :character_code, :region, :updated_at, :achievement_points, :portrait, :teams) 
```

Also, it might be useful to note that sc2ranks folder should be included in the ./lib folder and that you need to add a require 

```
require "./lib/sc2ranks/lib/sc2ranks.rb"
```

in order to get it working in a rails application. 

You can check out the application i'm building at sc2stars.com
